app/src/main/java/ch/epfl/sdp/appart/PanoramaActivity.java

```diff
@@ -152,16 +158,25 @@ private void loadImage() {
         PLSphericalPanorama panorama = new PLSphericalPanorama();
 
         CompletableFuture<Bitmap> bitmapFuture = new CompletableFuture<>();
-        database.accept(new GlideBitmapLoader(this, bitmapFuture, images.get(currImage)));
+
+        String imagePath = new StoragePathBuilder()
+                .toAdsStorageDirectory()
+                .toDirectory(currentAdId)
+                .withFile(images.get(currImage));
+
+        database.accept(new GlideBitmapLoader(this, bitmapFuture, imagePath));
 
         bitmapFuture.thenApply(bitmap -> {
+            hasCurrentImageLoadingFailed.complete(true);
             panorama.setImage(new PLImage(bitmap, true));
             panorama.getCamera().lookAtAndZoomFactor(30.0f, 90.0f, 0.5f, false);
             plManager.setPanorama(panorama);
             this.bitmap = bitmap;
             return bitmap;
         });
+
         bitmapFuture.exceptionally(e -> {
+            hasCurrentImageLoadingFailed.complete(false);
```

Comment:
If I understand correctly, hasCurrentImageLoadingFailed is completed to true when the loading doesn't fail and to false when it fails, is that correct ? If so, wouldn't it be more intuitive for it to work in the other way ? (As in, the boolean hasCurrentImageLoadingFailed is true when the loading fails) (I know i'm super nitpicking ahah)